src/input_parsers/OnnxParser.cpp

```diff
@@ -65,8 +65,8 @@ OnnxParser::OnnxParser( const String &path )
 void OnnxParser::generateQuery( InputQuery& query )
 {
     Set<String> inputNames = readInputNames();
-    String outputName = readOutputName();
-    processGraph( inputNames, outputName, query );
+    Set<String> terminalNames = readOutputNames();
```

Comment:
Why use terminal names instead of output names?

Reply:
I was trying to distinguish between the output nodes of the neural of the network, and the nodes that the query terminates at (that can be arbitrary nodes in the network). If you prefer the overloaded term, then I'm happy to switch back?

Reply:
Ah I see. In that case I think terminal names make sense. Could you please document this distinction between actual output and specified terminals somewhere?

Reply:
Other than this, the PR LGTM